### ISSUE TITLE
ConnectionPool repr fails when db is not specified

### ISSUE THREAD

**Issue by rancerbeta** [andymccurdy/redis-py#531]
This gives a key error because the repr for connectionpool uses the underlying connection object's repr string template. In this case, the connection object repr expects a db variable to be set because connection's init has a default db value of 0. There is no such thing in the connectionpool kwargs, so a key error is thrown when the string template is used.

```
import redis
r = redis.ConnectionPool(host='localhost', port=6379)
repr(r)
```

```
>>> x = redis.ConnectionPool(host='localhost', port=6379, max_connections=1)
>>> x.connection_kwargs
{'host': 'localhost', 'port': 6379}
>>> x.connection_class.description_format
'Connection<host=%(host)s,port=%(port)s,db=%(db)s>'
>>> x.connection_class.description_format % x.connection_kwargs
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
KeyError: 'db'
>>> 
```

**Comment by gyermolenko** [comment]
as I checked for #768 :
 no longer an issue - all format arguments are passed with defaults.

**Comment by andymccurdy** [comment]
This was resolved in redis-py 3.0.
